perf/DESIGN

```diff
@@ -0,0 +1,52 @@
+perf design
+===========
+
+The performance monitor is invoked as
+
+# kivaloo-perf -t <target socket> -s <source socket> -l <logfile> [-w secs]
+      [-p <pidfile>] [-1]
+
+It creates a socket at the address <source socket> on which it listens for
+incoming connections and accepts one at a time.  It opens a connection to
+<target socket> and forwards requests and responses back and forth.
+
+No attempt is made to confirm that requests are appropriate for the service
+being connected to or even that requests are valid; the user is responsible
+for ensuring this.  If the target service closes the connection, the daemon
+will exit (thus closing the connection it accepted).
+
+Performance statistics are written to <logfile> periodically, reporting for
+each type of request:
+ * The number of requests completed,
+ * The average time to complete,
```

Comment:
I'd use the word "mean" instead of "average".

Reply:
Agreed, it's one among many averages.

Reply:
Speaking of which, my disappointment is immeasurable and my day is ruined because this doesn't include the geothmetic meandian.